### ISSUE TITLE
Hint on scalability

### ISSUE THREAD

**Issue by TomLippincott** [knowledge-discovery/hw1#15]
@knowledge-discovery/cs365-fall-2016 
When interacting with the Retriever service (e.g. in your indexing implementation), bear in mind that asking for a million Communications at once is a *huge* chunk of data, and Thrift doesn't do streaming: so...perhaps if you avoid such massive requests, you can avoid issues with bigger data sets?

**Comment by anandsahuja** [comment]
Are we required to use the retriever service to read communications? The alternative being potentially modifying our eval.sh script to use something like the CommunicationReader class found in concrete-python concrete.util.file_io. I have not implemented this so I am unsure if it would be more efficient or would accomplish the same thing as the retriever service.

Sample usage:
        for (comm, filename) in CommunicationReader('multiple_comms.tar.gz'):
            do_something(comm)

**Comment by TomLippincott** [comment]
Yes, you should definitely use the retriever interface, and not modify the eval script.  The emphasis is on flexibility, and everyone is taking the same performance/efficiency hit of using the retriever service.

**Comment by lwolfsonkin** [comment]
So, is there any recommended block size to request communications?

**Comment by TomLippincott** [comment]
One at a time works fine for these data sets: e.g. for me, "eval.sh wikipedia_test.tgz" runs in 30s, "eval.sh twitter_test.tgz" in about 80s.  The larger data sets are more like ~30mins, but of course there's probably a sweet spot, possibly system-memory-determined, that would knock some time off that.  Feel free to experiment with different block sizes!
